Can't install go-gtk on Mac OSX Yosemite (10.10.4)

When I try to install it using the command `go get github.com/mattn/go-gtk/gtk`, I get the following error:

     # github.com/mattn/go-gtk/gdk
     ../../../golang/src/github.com/mattn/go-gtk/gdk/gdk_darwin.go:5:10: fatal error: 'gdk/gdkx.h' file not found
     #include <gdk/gdkx.h>
              ^
     1 error generated.

I've checked and pkg-config is correctly configured (I installed gtk+ with brew):

     $ pkg-config --cflags gdk-2.0
     -D_REENTRANT -I/opt/X11/include/cairo -I/opt/X11/include/pixman-1 -I/opt/X11/include -I/opt/X11/include/freetype2 -I/opt/X11/include/libpng15 -I/opt/X11/include -I/usr/local/Cellar/gtk+/2.24.28_2/include/gtk-2.0 -I/usr/local/Cellar/gtk+/2.24.28_2/lib/gtk-2.0/include -I/usr/local/Cellar/pango/1.36.8_1/include/pango-1.0 -I/usr/local/Cellar/gdk-pixbuf/2.30.8/include/gdk-pixbuf-2.0 -I/usr/local/Cellar/libpng/1.6.17/include/libpng16 -I/usr/local/Cellar/glib/2.44.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.44.1/lib/glib-2.0/include -I/usr/local/opt/gettext/include

and the files that it searches exist (gdk/gdkx.h).

Any suggestions on how to solve this ? Thanks a lot !

Of course, here it is:

    WORK=/var/folders/zc/sbpcg8qx7456m579jjjymlj00000gn/T/go-build733179269
    mkdir -p $WORK/github.com/mattn/go-gtk/gdk/_obj/
    mkdir -p $WORK/github.com/mattn/go-gtk/
    cd /Users/achedeuzot/golang/src/github.com/mattn/go-gtk/gdk
    pkg-config --cflags gdk-2.0 gthread-2.0
    pkg-config --libs gdk-2.0 gthread-2.0
    CGO_LDFLAGS="-g" "-O2" "-L/opt/X11/lib" "-L/usr/local/Cellar/gtk+/2.24.28_2/lib" "-L/usr/local/Cellar    /pango/1.36.8_1/lib" "-L/usr/local/Cellar/gdk-pixbuf/2.30.8/lib" "-L/usr/local/Cellar/glib/2.44.1/    lib" "-L/usr/local/opt/gettext/lib" "-lgdk-quartz-2.0" "-lpangocairo-1.0" "-lpango-1.0" "-lcairo" "-    lgdk_pixbuf-2.0" "-lgobject-2.0" "-lgthread-2.0" "-lglib-2.0" "-lintl" /usr/local/Cellar/go/1.4.2/    libexec/pkg/tool/darwin_amd64/cgo -objdir $WORK/github.com/mattn/go-gtk/gdk/_obj/ -- -D_REENTRANT -I/    opt/X11/include/cairo -I/opt/X11/include/pixman-1 -I/opt/X11/include -I/opt/X11/include/freetype2 -I/    opt/X11/include/libpng15 -I/opt/X11/include -I/usr/local/Cellar/gtk+/2.24.28_2/include/gtk-2.0 -I/usr    /local/Cellar/gtk+/2.24.28_2/lib/gtk-2.0/include -I/usr/local/Cellar/pango/1.36.8_1/include/pango-1.    0 -I/usr/local/Cellar/gdk-pixbuf/2.30.8/include/gdk-pixbuf-2.0 -I/usr/local/Cellar/libpng/1.6.17/    include/libpng16 -I/usr/local/Cellar/glib/2.44.1/include/glib-2.0 -I/usr/local/Cellar/glib/2.44.1/lib    /glib-2.0/include -I/usr/local/opt/gettext/include -I $WORK/github.com/mattn/go-gtk/gdk/_obj/ gdk.go     gdk_darwin.go
    # github.com/mattn/go-gtk/gdk
    ../../../golang/src/github.com/mattn/go-gtk/gdk/gdk_darwin.go:5:10: fatal error: 'gdk/gdkx.h' file     not found
    #include <gdk/gdkx.h>
             ^
    1 error generated.

I have this issue too:

go get -x github.com/mattn/go-gtk/gtk
WORK=/var/folders/kw/zqb4h35d5_5bcwb8b4sd_gc00000gn/T/go-build869683713
mkdir -p $WORK/github.com/mattn/go-gtk/gdk/_obj/
mkdir -p $WORK/github.com/mattn/go-gtk/
cd /Users/username/go/src/github.com/mattn/go-gtk/gdk
pkg-config --cflags gdk-2.0 gthread-2.0
pkg-config --libs gdk-2.0 gthread-2.0
CGO_LDFLAGS="-g" "-O2" "-L/usr/local/Cellar/gtk+/2.24.28_3/lib" "-L/usr/local/Cellar/pango/1.38.1/lib" "-L/usr/local/Cellar/cairo/1.14.4/lib" "-L/usr/local/Cellar/gdk-pixbuf/2.32.1/lib" "-L/usr/local/Cellar/glib/2.46.1_1/lib" "-L/usr/local/opt/gettext/lib" "-lgdk-quartz-2.0" "-lpangocairo-1.0" "-lpango-1.0" "-lcairo" "-lgdk_pixbuf-2.0" "-lgobject-2.0" "-lgthread-2.0" "-lglib-2.0" "-lintl" /usr/local/go/pkg/tool/darwin_amd64/cgo -objdir $WORK/github.com/mattn/go-gtk/gdk/_obj/ -importpath github.com/mattn/go-gtk/gdk -- -D_REENTRANT -I/usr/local/Cellar/gtk+/2.24.28_3/include/gtk-2.0 -I/usr/local/Cellar/gtk+/2.24.28_3/lib/gtk-2.0/include -I/usr/local/Cellar/pango/1.38.1/include/pango-1.0 -I/usr/local/Cellar/harfbuzz/1.0.4/include/harfbuzz -I/usr/local/Cellar/pango/1.38.1/include/pango-1.0 -I/usr/local/Cellar/cairo/1.14.4/include/cairo -I/usr/local/Cellar/pixman/0.32.8/include/pixman-1 -I/usr/local/Cellar/fontconfig/2.11.1/include -I/usr/local/Cellar/freetype/2.5.3_1/include/freetype2 -I/usr/local/Cellar/freetype/2.6_1/include/freetype2 -I/usr/local/Cellar/libpng/1.6.18/include/libpng16 -I/usr/local/Cellar/gdk-pixbuf/2.32.1/include/gdk-pixbuf-2.0 -I/usr/local/Cellar/libpng/1.6.18/include/libpng16 -I/usr/local/Cellar/glib/2.46.1_1/include/glib-2.0 -I/usr/local/Cellar/glib/2.46.1_1/lib/glib-2.0/include -I/usr/local/opt/gettext/include -I $WORK/github.com/mattn/go-gtk/gdk/_obj/ gdk.go gdk_darwin.go
# github.com/mattn/go-gtk/gdk
go/src/github.com/mattn/go-gtk/gdk/gdk_darwin.go:7:10: fatal error: 'gdk/gdkx.h' file not found
#include <gdk/gdkx.h>
         ^
1 error generated.

--- a/gdk/gdk_darwin.go
+++ b/gdk/gdk_darwin.go
@@ -1,1 +1,0 @@
-gdk_linux.go

--- a/gdk/gdk_quartz_darwin.go
+++ b/gdk/gdk_quartz_darwin.go
@@ -1,0 +1,33 @@
+// +build !with-x11
+
+package gdk
+
+/*
+#cgo pkg-config: gdk-2.0 gthread-2.0
+#cgo CFLAGS: -x objective-c
+
+#include <gdk/gdk.h>
+#include <gdk/gdkquartz.h>
+
+// Must return void* to avoid "struct size calculation error off=8 bytesize=0"
+// See:
+// - https://github.com/golang/go/issues/12065
+// - http://thread0.me/2015/07/gogoa-cocoa-bindings-for-go/
+void* getNsWindow(GdkWindow *w) {
+	return (void*) gdk_quartz_window_get_nswindow(w);
+}
+
+*/
+import "C"
+
+import "unsafe"
+
+type NSWindow struct {
+	ID unsafe.Pointer
+}
+
+func (v *Window) GetNativeWindow() *NSWindow {
+	return &NSWindow{
+		unsafe.Pointer(C.getNsWindow(v.GWindow)),
+	}
+}

--- a/gdk/gdk_x11_darwin.go
+++ b/gdk/gdk_x11_darwin.go
@@ -1,0 +1,16 @@
+// +build with-x11
+
+package gdk
+
+/*
+#cgo pkg-config: gdk-2.0 gthread-2.0
+
+#include <gdk/gdk.h>
+#include <gdk/gdkx.h>
+*/
+import "C"
+import "unsafe"
+
+func (v *Window) GetNativeWindowID() int32 {
+	return int32(C.gdk_x11_drawable_get_xid((*C.GdkDrawable)(unsafe.Pointer(v.GWindow))))
+}

--- a/gtk/gtk.go
+++ b/gtk/gtk.go
@@ -1409,10 +1409,6 @@ func (v *Window) Resize(width int, height int) {
 	C.gtk_window_resize(WINDOW(v), gint(width), gint(height))
 }
 
-func (v *Window) XID() int32 {
-	return gdk.WindowFromUnsafe(unsafe.Pointer(v.GWidget.window)).GetNativeWindowID()
-}
-
 // gtk_window_set_default_icon_list
 // gtk_window_set_default_icon
 // gtk_window_set_default_icon_from_file

--- a/gtk/gtk.go.h
+++ b/gtk/gtk.go.h
@@ -6,7 +6,9 @@
 #ifdef _WIN32
 #include <windows.h>
 #else
-#include <gdk/gdkx.h>
+//Why is this necessary?
+//#include <gdk/gdkx.h>
+//#include <gdk/gdkquartz.h>
 #endif
 #include <unistd.h>
 #include <stdlib.h>

--- a/gtk/gtk_x11.go
+++ b/gtk/gtk_x11.go
@@ -1,0 +1,13 @@
+// +build with-x11
+
+package gtk
+
+import (
+	"unsafe"
+
+	"github.com/mattn/go-gtk/gdk"
+)
+
+func (v *Window) XID() int32 {
+	return gdk.WindowFromUnsafe(unsafe.Pointer(v.GWidget.window)).GetNativeWindowID()
+}

